packages/@aws-cdk/aws-ecs/lib/cluster.ts

```diff
@@ -245,7 +266,22 @@ export class Cluster extends Resource implements ICluster {
     // since it's harmless, but we'd prefer not to add unexpected new
     // resources to the stack which could surprise users working with
     // brown-field CDK apps and stacks.
-    Aspects.of(this).add(new MaybeCreateCapacityProviderAssociations(this, id, this._capacityProviderNames));
+    const _this = this;
+    Aspects.of(this).add({
+      visit(node: IConstruct): void {
+        if (node.node.addr === _this.node.addr) {
+          if ((_this._defaultCapacityProviderStrategy.length > 0 || _this._capacityProviderNames.length > 0)) {
+            if (!_this.node.tryFindChild(id)) {
+              new CfnClusterCapacityProviderAssociations(_this, id, {
+                cluster: _this.clusterName,
+                defaultCapacityProviderStrategy: _this._defaultCapacityProviderStrategy,
+                capacityProviders: _this._capacityProviderNames,
+              });
+            }
+          }
+        }
+      },
+    });
```

Comment:
After discussed with CDK expert  , Here I removed class, Lazy list and added above mentioned function in Aspect inline, this is because 
Actually we probably do still need to use lazy with both properties. Aspects get called before things are resolved
It works in the test because the aspect is invoked after we add the capacity providers, but if we added some lazily it would probably fail.
Deleting MaybeCreateCapacityProviderAssociation class and adding new _this to aspect:  For this I’m not 100% sure, but I think it has something to do with the fact that we pass the object to the class and then after that we update the object. So I tried that and it works.

Reply:
Why doesn't it work with the class, using this exact logic? I need you to find out whether it would, and whether this current implementation works. I don't see a test for that.

Reply:
This conversation doesn't seem to be resolved.

Reply:
Looks like we were able to get it working with the class. This clarifies things in the code enough for me to resolve this.